src/backend/app/upload_basemaps_s3.py

```diff
@@ -0,0 +1,134 @@
+"""Upload basemaps on the filesystem to the connected S3 bucket."""
+
+import asyncio
+from pathlib import Path
+
+from dotenv import load_dotenv
+from psycopg import AsyncConnection
+
+from app.config import settings
+from app.db.models import DbBackgroundTask, DbBasemap, DbProject
+from app.projects.project_schemas import BasemapOut, BasemapUpdate
+from app.s3 import add_file_to_bucket
+
+# Load environment variables
+load_dotenv(Path(__file__).parent.parent / ".env")
+
+
+async def upload_basemap_file(
+    project_id: int, org_id: int, basemap: DbBasemap, filepath: str
+) -> str:
+    """Upload the file to S3 and cleanup the file after."""
+    try:
+        basemap_out = BasemapOut(
+            **basemap.model_dump(exclude={"url"}),
+            url=filepath,
+        )
+        basemap_s3_path = (
+            f"{org_id}/{project_id}/basemaps/{basemap_out.id}.{basemap_out.format}"
+        )
+        print(f"Uploading basemap to S3 path: {basemap_s3_path}")
+        add_file_to_bucket(
+            settings.S3_BUCKET_NAME,
+            basemap_s3_path,
+            filepath,
+            content_type=basemap_out.mimetype,
+        )
+        basemap_external_s3_url = (
+            f"{settings.S3_DOWNLOAD_ROOT}/{settings.S3_BUCKET_NAME}/{basemap_s3_path}"
+        )
+        print(f"Upload of basemap to S3 complete: {basemap_external_s3_url}")
+        # Delete file on disk
+        Path(filepath).unlink(missing_ok=True)
+
+        return basemap_external_s3_url
+    except Exception as e:
+        print(f"Error uploading basemap file: {e}")
+        raise
+
+
+async def basemap_files_to_s3():
+    """Upload basemap file & update the DbBasemap.url field."""
+    async with await AsyncConnection.connect(
+        settings.FMTM_DB_URL.unicode_string(),
+    ) as db:
+        try:
+            projects = await DbProject.all(db)
+            if not projects:
+                print("No projects found.")
+                return
+
+            for project in projects:
+                print("")
+                print("--------------------------")
+                print(f"------- Project {project.id} -------")
+                print("--------------------------")
+                print("")
```

Comment:
Are these print statements necessary here?

Reply:
Yep!

It's a migration to run on command line, so you get an output like this:

![image](https://github.com/user-attachments/assets/781d96e8-f501-4f50-abd1-8089ea353265)